### PR TITLE
Increased stack size for migraphx-hiprtc-driver to 4MB on Windows

### PR DESCRIPTION
**Pull request by slojosic-amd**
Default 1MB stack size is not enough for migraphx-hiprtc-driver.exe on Windows.
@apwojcik

### CODE REVIEW

**Review by umangyadav — APPROVED** [review]

File: src/targets/gpu/hiprtc/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,9 @@ add_executable(migraphx-hiprtc-driver
     main.cpp
 )
 rocm_clang_tidy_check(migraphx-hiprtc-driver)
+if(WIN32)
+    target_link_options(migraphx-hiprtc-driver PRIVATE -Xlinker /stack:4194304)
```
Comment (pfultz2) [inline src/targets/gpu/hiprtc/CMakeLists.txt:30]: I think this option is only valid when using MSVC linker. We should probably check for that.

Comment (apwojcik) [reply src/targets/gpu/hiprtc/CMakeLists.txt:30]: There is no need. The llvm-ld.exe is just a frontend, and on Windows, it invokes the MSVC linker (link.exe).

Comment (apwojcik) [reply src/targets/gpu/hiprtc/CMakeLists.txt:30]: The only exception is when MINGW (or CYGWIN) is used then the GNU linker is used.

Comment (pfultz2) [reply src/targets/gpu/hiprtc/CMakeLists.txt:30]: Sure, but we could add a check for GCC and skip adding this flag so it wont break any users that might want to build with gcc in the future.

Comment (apwojcik) [reply src/targets/gpu/hiprtc/CMakeLists.txt]: @pfultz2 done

**Pull request merged**